addons/mail/static/src/discuss/call/common/rtc_service.js

```diff
@@ -543,7 +618,106 @@ export class Rtc {
+                    const session = this.store.RtcSession.get(Number(id));
+                    if (!session) {
+                        return;
+                    }
+                    // `isRaisingHand` is turned into the Date `raisingHand`
+                    this.setRemoteRaiseHand(session, info.isRaisingHand);
+                    delete info.isRaisingHand;
+                    Object.assign(session, info);
+                }
+                return;
+            case "track":
+                {
+                    const { sessionId, type, track, active } = payload;
+                    const session = this.store.RtcSession.get(sessionId);
+                    if (!session) {
+                        return;
+                    }
+                    try {
+                        await this.handleRemoteTrack({ session, track, type, active });
+                    } catch {
+                        // ignored, the session may be closing.
+                        // this can happen when you join a call from another tab in which you have another session.
+                    }
+                    // makes sure we are not downloading a video that is not displayed
+                    setTimeout(() => {
+                        this.updateVideoDownload(session);
+                    }, 2000);
+                }
+                return;
+        }
+    }
+
+    async _handleSfuClientStateChange({ detail: { state, cause } }) {
+        this.state.serverState = state;
+        switch (state) {
+            case this.SFU_CLIENT_STATE.AUTHENTICATED:
+                // if we are hot-swapping connection type, we clear the p2p as late as possible
+                this.clearPeerToPeer();
+                break;
+            case this.SFU_CLIENT_STATE.CONNECTED:
+                this.sfuClient.updateInfo(this.formatInfo(), {
+                    needRefresh: true, // asks the server to send the info from all the channel
+                });
+                this.sfuClient.updateUpload("audio", this.state.audioTrack);
+                this.sfuClient.updateUpload("camera", this.state.cameraTrack);
+                this.sfuClient.updateUpload("screen", this.state.screenTrack);
+                return;
+            case this.SFU_CLIENT_STATE.CLOSED:
+                {
+                    let text;
+                    if (cause === "full") {
+                        text = _t("Channel full");
+                    } else {
+                        text = _t("Connection to SFU server closed by the server");
+                    }
+                    this.notification.add(text, {
+                        type: "warning",
+                    });
+                    await this.leaveCall();
+                }
+                return;
+        }
+    }
+
+    async call() {
+        if (this.state.connectionType === CONNECTION_TYPES.SERVER) {
+            if (this.sfuClient.state === this.SFU_CLIENT_STATE.DISCONNECTED) {
```

Comment:
merge with the outer if?

Reply:
or put the early return in advance to reduce the indent changes and easier to read. `if (this.state.connectionType !== CONNECTION_TYPES.SERVER) { return; }`